MVCapplication/src/main/java/com/softserve/if072/mvcapp/service/ShoppingListService.java

```diff
@@ -0,0 +1,139 @@
+package com.softserve.if072.mvcapp.service;
+
+import com.softserve.if072.common.model.Product;
+import com.softserve.if072.common.model.ShoppingList;
+import com.softserve.if072.common.model.Unit;
+import com.softserve.if072.common.model.User;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+/**
+ * This class contains methods for obtaining and changing a shopping list.
+ *
+ * @author Oleh Pochernin
+ */
+@Service
+public class ShoppingListService {
+    private static final Logger LOG = LogManager.getLogger(ShoppingListService.class);
+    private static final String INFO_LOG_TEMPLATE = "Shopping list element (userId: %d, productId: %d) has been %s.";
+
+    @Value("${service.shoppingList}")
+    private String shoppingListUrl;
+
+    @Value("${service.shoppingList.byUser}")
+    private String shoppingListByUserUrl;
+
+    @Value("${service.shoppingList.byUserAndProduct}")
+    private String shoppingListByUserAndProductUrl;
+
+    @Value("${service.product.id}")
+    private String productById;
+
+    @Autowired
+    RestTemplate restTemplate;
+
+    /**
+     * This method receives all shopping list's elements from RESTful service and returns them.
+     *
+     * @param userId id of user whose shopping list must be received
+     * @return all shopping list's elements for user
+     */
+    public List<ShoppingList> getAllElements(int userId) {
+        List<ShoppingList> shoppingList = restTemplate.getForObject(String.format(shoppingListByUserUrl, userId), List.class);
+
+        if (CollectionUtils.isNotEmpty(shoppingList)) {
+            LOG.info(String.format("Shopping list with %d elements has been received.", shoppingList.size()));
```

Comment:
try feature of the log4j 2
`LOG.info("Shopping list with {} elements has been received.", shoppingList.size());`